feat(i18n): extract text from admin form responses

Jaimeloeuf opened the pull request:
Extract text from features/admin-form/responses/**

There are some that are more tricky, to be done, and tracked by https://github.com/opengovsg/FormSG/issues/8204

Jaimeloeuf committed:
feat(i18n): extract text from admin form responses

littlemight commented on frontend/src/features/admin-form/responses/ChartsPage/ChartsPage.tsx:
```diff
@@ -33,8 +35,7 @@ export const ChartsPage = (): JSX.Element => {
     toast({
-      description:
-        'There was an error retrieving your form. Please try again later.',
+      description: t('features.adminForm.toasts.form.retrival.error'),
```
nit: typo, should be `retrieval`

littlemight commented on frontend/src/features/admin-form/responses/ResponsesPage/storage/UnlockedResponses/UnlockedResponses.tsx:
```diff
@@ -65,8 +59,14 @@ export const UnlockedResponses = (): JSX.Element => {
+              </Text>{' '}
+              {submissionId
+                ? t(
+                    'features.adminForm.responses.responsesPage.storage.unlockedResponses.unlockedResponses.resultsFound',
```
for every `import simplur` removed. make sure the equivalent JSON translation file takes care of plurals as well. in this example, seems like it deferred to `result(s) found`.

you can use [ICU format](https://react.i18next.com/misc/using-with-icu-format)
working example on codebase [here](https://github.com/opengovsg/FormSG/blob/1340d28e7ca58c7f39e366fdbd598358957a5750/frontend/src/i18n/locales/features/public-form/table/en-sg.ts#L7-L8)

Jaimeloeuf replied:
understood, am wondering for some texts that are already for eg, result(s) do you want me to convert these to ICU format too?

littlemight replied:
if the existing copy is already like that, I think safe to leave it be.
if the existing copy explicitly cares for single/plurals, then keep it the same

LoneRifle replied:
rule of thumb would be to minimise diffs on Chromatic for ease of review, given the number of files we have to work through

littlemight commented on frontend/src/features/admin-form/responses/ResponsesPage/storage/UnlockedResponses/ProgressModal/CompleteScreen.tsx:
```diff
@@ -31,9 +32,11 @@ export const CompleteScreen = ({
   const isMobile = useIsMobile()
   const mdComponents = useMdComponents()
 
+  // TODO(#8204) Extract out text after removing simplur and maybe with ICU format
```
(not review): let me collate all the todos on the issue as well. it might be a common symptom across the codebase

littlemight commented on frontend/src/features/admin-form/responses/ResponsesPage/storage/UnlockedResponses/ResponsesTable/ResponsesTable.tsx:
```diff
@@ -63,38 +64,51 @@ const StatusBadge = ({
+  )
+}
 
+// TODO(#8204) Extract out text after moving into component to use the hook
```
ditto: write on issue

littlemight commented on frontend/src/features/admin-form/responses/constants.ts:
```diff
@@ -1,3 +1,4 @@
+// TODO(#8204) Use text labels in i18n files
```
ditto: write on issue

littlemight commented on frontend/src/features/admin-form/responses/ResponsesPage/storage/UnlockedResponses/DownloadWithAttachmentModal/ConfirmationScreen.tsx:
```diff
@@ -45,52 +46,80 @@ export const ConfirmationScreen = ({
       <ModalBody whiteSpace="pre-wrap" color="secondary.500">
         <Stack spacing="1rem">
           <Text>
+            {/* TODO(#8204) Maintain bold when extracting */}
```
ditto: write on issue

littlemight commented on frontend/src/features/admin-form/responses/ChartsPage/UnlockedCharts/UnlockedChartsContainer.tsx:
```diff
@@ -156,7 +151,9 @@ export const UnlockedChartsContainer = () => {
             </Text>
-            {prettifiedResponsesCount}
+            {t(
+              'features.adminForm.responses.charts.unlockedChartsContainer.responsesRetrieved',
```
ditto: simplur

LoneRifle replied:
you can't really use simplur or ICU plurals here, since the value is in a separate element for bolding

littlemight commented on frontend/src/features/admin-form/responses/FeedbackPage/FeedbackPage.tsx:
```diff
@@ -241,34 +327,34 @@ const getReviewInformationComponent = (
+            {props.count}
           </Text>
-          {simplur` ${[count || 0]}review[|s] to date`}
+          {t('features.adminForm.responses.feedbackPage.review.reviewsToDate')}
```
ditto: simplur

littlemight commented on frontend/src/features/admin-form/responses/FeedbackPage/FeedbackPage.tsx:
```diff
@@ -241,34 +327,34 @@ const getReviewInformationComponent = (
+          {props.count}
         </Text>
-        {simplur` ${[count || 0]}issue[|s] to date`}
+        {t('features.adminForm.responses.feedbackPage.issue.issuesToDate')}
```
ditto: simplur

littlemight commented on frontend/src/features/admin-form/responses/IndividualResponsePage/IndividualResponsePage.tsx:
```diff
@@ -243,15 +247,20 @@ export const IndividualResponsePage = (): JSX.Element => {
                   }
                 >
-                  {simplur`Download ${attachmentDownloadUrls.size} attachment[|s] as .zip`}
+                  {t(
```
ditto: simplur

littlemight commented on frontend/src/features/admin-form/responses/ResponsesPage/email/EmailResponsesTab.tsx:
```diff
@@ -23,11 +25,15 @@ export const EmailResponsesTab = (): JSX.Element => {
               {responsesCount?.toLocaleString() ?? '-'}
             </Text>
-            {simplur` ${[responsesCount ?? 0]}response[|s] to date`}
+            {t(
```
ditto: simplur

littlemight commented on frontend/src/features/admin-form/responses/components/SecretKeyVerification/SecretKeyVerification.tsx:
```diff
@@ -29,15 +31,19 @@ export const SecretKeyVerification = ({
               </Text>
-              {simplur` ${[totalResponsesCount ?? 0]}response[|s] to date`}
+              {t(
+                'features.adminForm.responses.components.secretKeyVerification.responsesToDate',
```
ditto: simplur

littlemight commented on frontend/src/i18n/locales/features/admin-form/responses/charts/en-sg.ts:
```diff
@@ -0,0 +1,47 @@
+    },
+  },
+  unlockedChartsContainer: {
+    responsesRetrieved: ' response(s) retrieved',
```
nit: avoid leading or trailing whitespace in locale files (?)
hmm this is just my early hunch, I'm slightly leaning to not having leading/trailing whitespace and handle it in code instead (for visibility/debugging). 

on the other hand, I'm not sure what the best practice is: maybe having it in the locale files could actually be beneficial?

sir @LoneRifle do weigh in your thoughts as you've extracted a bunch of texts previously 🙏

Jaimeloeuf replied:
makes sense, i think i was just copying the fact that the original text had leading/trailing whitespace, so its in the i18n files as i just copied them in wholesale

LoneRifle replied:
yea let's try to avoid trailing/leading whitespace in the locale values. See eb11d56 as a reference for how whitespace should be handled

littlemight reviewed (commented):
haven't dissect in detail yet, but left few remarks as a first cut of the review

LoneRifle committed:
fix(i18n): act on review comments

- Keep whitespace to codebase
- Use ICU for plurals
- Move TODO comments to issue

LoneRifle committed:
refactor(i18n): remove feedback-page, covered by feedback
